### DISCUSSION TITLE
https://auth.tesla.com/oauth2/v3/token NOT returning a refresh_token

### DISCUSSION THREAD

**Discussion by SylvainGa** [timdorr/tesla-api#551] [Q&A]
Hi, still struggling with this. I'm trying to refresh an access token by calling https://auth.tesla.com/oauth2/v3/token with the following parameters:
```
"grant_type" => "refresh_token",
"client_id" => "ownerapi",
"refresh_token" => token,  <- Variable holding the refresh token
"scope" => "openid email offline_access"
 ```

But this is what is returned
```
{
 id_token=> LongTokenReturned
 access_token=> LongTokenReturned
 token_type=>Bearer
 expires_in=>28800
}
```
It's not returning me a refresh token so in 8 hours, I can't get a new access token :-( What am I doing wrong? Thanks.

**Comment by haha150** [comment]
This behaviour is new and correct, see https://github.com/timdorr/tesla-api/issues/548, that is the refresh token

**Reply by SylvainGa** [reply]
Sorry but no, this is not the same. They"re getting an invalid request. Me, I had a result code of 200 but as you can see from the initial post, the refresh token wasn't returned. Anyway, it might have been a transient error because it's working now. I'm getting both the Access and Refresh tokens from the call.
